src/garmin.c

```diff
@@ -156,11 +186,23 @@ make_space(struct file_list *files)
 		array = realloc(files->array, new_size);
 		if (!array)
 			return DC_STATUS_NOMEMORY;
-
+#ifdef HAVE_LIBMTP
+		if (is_mtp) {
+			unsigned int *mtp_ids;
+			new_size = n * sizeof(unsigned int);
+			mtp_ids = realloc(files->mtp_ids, new_size);
+			if (!mtp_ids)
+				return DC_STATUS_NOMEMORY;
+			files->mtp_ids = mtp_ids;
+		}
+#else
+		(void)is_mtp;
+#endif
```

Comment:
Hmm.

I'd just unconditionally add "unsigned int mtp_id" to "struct fit_name" instead. The non-mtp case can just set it to zero and ignore it.

You'd have to make the "name_cmp()" function actually do something like


    static int name_cmp(const void *_a, const void *_b)
    {
        const struct fit_name *a = _a;
        const struct fit_name *b = _b;

        // Sort reverse string ordering (newest first), so use 'b,a'
        return strcmp(b->name, a->name);
    }

and maybe there is something else that depends on "struct fit_name" only being the name, but that would seem to be a cleanup regardless.

Reply:
ok, I'll look into that